apostrophecms/apostrophe
# Pro 2840 webpack extensions options

[PRO-2840](https://linear.app/apostrophecms/issue/PRO-2840/[michelin-p1]-ability-to-pass-options-to-a-webpack-extension)

## Summary

Adds possibility to pass options from any module to a webpack extension.
Feature has been discussed and tested directly with Jonathan.
`extensionOptions` can be an object or a function, and don't need to specifically re merge options inside because we do it in a reducer.
Tests have been added.

Documentation in this PR: https://github.com/apostrophecms/a3-docs/pull/211

## What are the specific steps to test this change?

* Create webpack extensions that are functions taking an options parameter
* Create webpack `extensionsOptions` in multiple modules
* Verify that they are nicely merged together.
* Verify that options are passes to the extension
* Verify that merged `extensionsOptions` are stored in  `self.apos.asset.webpackExtensionOptions`


## What kind of change does this PR introduce?


## Make sure the PR fulfills these requirements:
